src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -480,6 +480,10 @@ void QgsExpressionCalculatorLocatorFilter::fetchResults( const QString &string,
   context << QgsExpressionContextUtils::globalScope()
           << QgsExpressionContextUtils::projectScope( QgsProject::instance() );
 
+  QgsVectorLayer *layer = qobject_cast< QgsVectorLayer *>( QgisApp::instance()->activeLayer() );
```

Comment:
No need to cast to vector here -- raster and mesh layers are ok.

Reply:
I was wondering about that. OK.